src/packagedcode/spec.py

```diff
@@ -0,0 +1,156 @@
+        """
+        Return dictionary contains podspec or gemspec file data.
+        """
+        with io.open(location, encoding='utf-8', closefd=True) as data:
+            lines = data.readlines()
+
+        spec_data = {}
+
+        for line in lines:
+            if '.name' in line:
+                name = re.sub(r'/*.*name.*?=', '', line)
+                spec_data['name'] = get_stripped_data(name)
+            elif '.version' in line and '.version.' not in line:
+                version = re.sub(r'/*.*version.*?=', '', line)
+                spec_data['version'] = get_stripped_data(version)
+            elif '.license' in line:
+                license_type = re.sub(r'/*.*license.*?=', '', line)
+                spec_data['license'] = get_stripped_data(license_type)
+            elif '.source' in line and '.source_files' not in line:
+                source = re.sub(r'/*.*source.*?>', '', line)
+                stripped_source = re.sub(r',.*', '', source)
+                spec_data['source'] = get_stripped_data(stripped_source)
+            elif '.email' in line:
+                emails = re.sub(r'/*.*email.*?=', '', line)
+                stripped_emails = get_stripped_data(emails)
+                stripped_emails = stripped_emails.strip()
+                stripped_emails = stripped_emails.split(',')
+                spec_data['email'] = stripped_emails
+            elif '.author' in line:
+                authors = re.sub(r'/*.*author.*?=', '', line)
+                stripped_authors = get_stripped_data(authors)
+                stripped_authors = stripped_authors.replace(' => ', "=>")
+                stripped_authors = stripped_authors.strip()
+                stripped_authors = stripped_authors.split(',')
+                spec_data['author'] = stripped_authors
+            elif '.summary' in line:
+                summary = re.sub(r'/*.*summary.*?=', '', line)
+                spec_data['summary'] = get_stripped_data(summary)
+            elif '.description' in line:
+                if location.endswith('.gemspec'):
+                    desc = re.sub(r'/*.*description.*?=', '', line)
+                    spec_data['description'] = get_stripped_data(desc)
+                else:
+                    spec_data['description'] = get_description(location)
+            elif '.homepage' in line:
+                homepage_url = re.sub(r'/*.*homepage.*?=', '', line)
+                spec_data['homepage_url'] = get_stripped_data(homepage_url)
+
+        parser = GemfileParser(location)
+        deps = parser.parse()
+        dependencies = OrderedDict()
+        for key in deps:
+            if deps[key]:
+                for dependency in deps[key]:
+                    dependencies[dependency.name] = (dependency.requirement).split(',')
+        spec_data['dependencies'] = dependencies
+
+        return spec_data
+
+
+def get_stripped_data(line):
+    """
+    Return data after removing unnecessary special character and space.
```

Comment:
is this a line? or data? or a string?

Reply:
This will be a line. I will update that in the next changes :)